lib/matplotlib/widgets.py

```diff
@@ -1066,12 +1069,21 @@ def _rendercursor(self):
 
         text = self.text_disp.get_text()  # Save value before overwriting it.
         widthtext = text[:self.cursor_index]
+
+        bb_1 = self.text_disp.get_window_extent()
         self.text_disp.set_text(widthtext or ",")
-        bb = self.text_disp.get_window_extent()
-        if not widthtext:  # Use the comma for the height, but keep width to 0.
-            bb.x1 = bb.x0
+        bb_2 = self.text_disp.get_window_extent()
+
+        if bb_1.y0 == bb_1.y1:  # no text
+            bb_1.y0 -= (bb_2.y1-bb_2.y0)/2
+            bb_1.y1 += (bb_2.y1-bb_2.y0)/2
+        elif not widthtext:  # cursor at index 0
+            bb_1.x1 = bb_1.x0
+        else:
+            bb_1.x1 = bb_1.x0 + (bb_2.x1 - bb_2.x0)
+
         self.cursor.set(
-            segments=[[(bb.x1, bb.y0), (bb.x1, bb.y1)]], visible=True)
+            segments=[[(bb_1.x1, bb_1.y0), (bb_1.x1, bb_1.y1)]], visible=True)
```

Comment:
I don't follow this code.  `bb` is now `bb_2`, but the old usages of `bb` are replaced by `bb_1`?!?

You should explain what you do here. Also, names `bb_1`, `bb_2` could be more descriptive.

Reply:
What do you think about it now? I tried to use more descriptive names bb_1 -> bb_text and bb_2 -> nn_widthtext and comment a bit